src/lang/modifyAst/addFillet.ts

```diff
@@ -0,0 +1,182 @@
+import {
+  ArrayExpression,
+  CallExpression,
+  ObjectExpression,
+  PathToNode,
+  PipeExpression,
+  Program,
+  VariableDeclaration,
+  VariableDeclarator,
+} from '../wasm'
+import {
+  createCallExpressionStdLib,
+  createLiteral,
+  createPipeSubstitution,
+  createObjectExpression,
+  createArrayExpression,
+  findUniqueName,
+} from '../modifyAst'
+import { getNodeFromPath } from '../queryAst'
+import { KCL_DEFAULT_CONSTANT_PREFIXES } from 'lib/constants'
+
+export function addFilletToEdgeWithinExtrude(
+  node: Program,
+  pathToSegmentNode: PathToNode,
+  pathToExtrudeNode: PathToNode,
+  filletRadius: number = 5
+): { modifiedAst: Program; pathToSegmentNode: PathToNode } {
```

Comment:
if I would just add an Error as a possible return type:
`{ modifiedAst: Program; pathToSegmentNode: PathToNode } | Error`
then I would have to return an Error instead of promise: 
`return new Error(...)`

if we want to return the Promise, like this:
`return Promise.reject(...)`
then I would need to wrap the whole function in the promise:
```ts
Promise<{ modifiedAst: Program; pathToSegmentNode: PathToNode } | Error> {
  return new Promise((resolve, reject) => {
```
is this the way ? 
![w69KswqfwebHt4TiZ4HmCd](https://github.com/KittyCAD/modeling-app/assets/156543465/472f5c86-d487-489f-b563-8df78d337f60)

Reply:
You may return an `Error` directly without the promise if that's what makes sense. Promise.reject(new Error(...)) is just for functions which are async.